gradle-guide-error-prone/src/main/java/com/palantir/gradle/guide/errorprone/besteffort/ProviderImprovements.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * (c) Copyright 2025 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.gradle.guide.errorprone.besteffort;
+
+import com.google.auto.service.AutoService;
+import com.google.errorprone.BugPattern;
+import com.google.errorprone.BugPattern.SeverityLevel;
+import com.google.errorprone.VisitorState;
+import com.google.errorprone.bugpatterns.BugChecker;
+import com.google.errorprone.matchers.Description;
+import com.google.errorprone.matchers.Matcher;
+import com.google.errorprone.matchers.Matchers;
+import com.palantir.gradle.guide.errorprone.GradleGuideBugChecker;
+import com.palantir.gradle.guide.errorprone.utils.ReplacementTracker.SuggestedFixBuilder;
+import com.palantir.gradle.guide.errorprone.utils.TreeUtils;
+import com.sun.source.tree.ExpressionTree;
+import com.sun.source.tree.LambdaExpressionTree;
+import com.sun.source.tree.MemberSelectTree;
+import com.sun.source.tree.MethodInvocationTree;
+
+@AutoService(BugChecker.class)
+@BugPattern(
+        severity = SeverityLevel.SUGGESTION,
+        summary = "Do not call `Provider.get`. Instead, pass providers directly to methods that accept them, "
+                + "or transform providers using `Provider.map` or `Provider.flatMap`, or combine providers using "
+                + "`Provider.zip`. Calling `Provider.get` causes Gradle to lose track of implicit dependencies and "
+                + "can lead to timing issues by reading values too early.")
+public final class ProviderImprovements extends GradleGuideBugChecker
+        implements BugChecker.MethodInvocationTreeMatcher {
+
+    private static final Matcher<ExpressionTree> MATCHER = Matchers.instanceMethod()
+            .onDescendantOf("org.gradle.api.provider.Provider")
+            .namedAnyOf("get", "getOrNull", "getOrElse");
```

Comment:
not sure what you'd do in this instance, but I think `isPresent` is also something you want to avoid at configuration time where possible

Reply:
This is vestigial from when I was actually erroring on `Provider.get/getOrNull` etc (which doesn't work without understanding the context it's being called it) - reduced it back to just `Provider.get` as that's what we handle so far.